src/picomsg/src/oper_db_writer.cpp

```diff
@@ -25,6 +25,82 @@ static bool terminator_state_to_oper_db(const pm_terminator_conn_state terminato
 	return false;
 }
 
+static bool tunnel_status_to_oper_db(const uint16_t tunnel_status, enum LIBOPER_DB_CLIENT_FIELD_ENUM_VALUES& oper_db_value) {
+	switch(tunnel_status) {
+		case IKE_CREATED:
+			oper_db_value = LIBOPER_DB_CLIENT_STATUS_IKE_CREATED;
+			return true;
+		case IKE_CONNECTING:
+			oper_db_value = LIBOPER_DB_CLIENT_STATUS_IKE_CONNECTING;
+			return true;
+		case IKE_ESTABLISHED:
+			oper_db_value = LIBOPER_DB_CLIENT_STATUS_IKE_ESTABLISHED;
+			return true;
+		case IKE_REKEYING:
+			oper_db_value = LIBOPER_DB_CLIENT_STATUS_IKE_REKEYING;
+			return true;
+		case IKE_REKEYED:
+			oper_db_value = LIBOPER_DB_CLIENT_STATUS_IKE_REKEYED;
+			return true;
+		case IKE_DELETING:
+			oper_db_value = LIBOPER_DB_CLIENT_STATUS_IKE_DELETING;
+			return true;
+		default:
+			PICOMSG_LOG(PICOMSG_LOG_ERR, "got invalid ike state %d", tunnel_status);
+			return false;
+	}
+
+	return false;
+}
+
+static bool version_to_oper_db(const uint32_t version, enum LIBOPER_DB_CLIENT_FIELD_ENUM_VALUES& oper_db_value) {
+	switch(version) {
+		case 1:
+				oper_db_value = LIBOPER_DB_CLIENT_IKE_VERSION_1;
+				return true;
+		case 2:
+				oper_db_value = LIBOPER_DB_CLIENT_IKE_VERSION_2;
+				return true;
+	}
+	PICOMSG_LOG(PICOMSG_LOG_ERR, "got invalid ike version %u", version);
+	return false;
+}
+
+static bool enc_algo_to_oper_db(const uint16_t enc_alg, const uint16_t key_size, enum LIBOPER_DB_CLIENT_FIELD_ENUM_VALUES& oper_db_value) {
+	switch(enc_alg) {
+		case ENCR_AES_CBC:
+			switch(key_size) {
+				case 128:
+					oper_db_value = LIBOPER_DB_CLIENT_ENC_ALG_AES_128_CBC;
+					return true;
+				case 256:
+					oper_db_value = LIBOPER_DB_CLIENT_ENC_ALG_AES_256_CBC;
+					return true;
+				PICOMSG_LOG(PICOMSG_LOG_ERR, "got invalid encryption algo key size %hu", key_size);
```

Comment:
Also, include in the log that got a wrong key size specifically for AES-CBC encryption